### ISSUE TITLE
Hierarchical training and reward set

### ISSUE THREAD

**Issue by JakobSchmitt** [rail-berkeley/softlearning#82]
Hi, 
I found your paper "Latent Space Policies for Hierarchical Reinforcement Learning" very interesting and was glad you published the code. Motivated by your results, I'd like to implement the ant maze with hierarchical policies and compound skills / different rewards. 
I didn't come up with an answer to the following questions. It would be great if you could help me out!

I assume that I have to pretrain a lower level policy first.  How do I freeze the low level weights in the next step and how can I add a high level policy on top?

In the paper you mentioned a set of K reward functions.  Where can I define the reward set ?

Thank you!

**Comment by hartikainen** [comment]
Hey @JakobSchmitt, thanks for your interest! I left some answers inline below. One thing to note is that this repo doesn't currently support the training of hierarchical RealNVP policies. If you want to train hierarchical policies, I would recommend checking out https://github.com/haarnoja/sac even though it is no longer being maintained. It's possible that we'll have some hierarchical policies in this repo at some point in the future, but I don't have a good estimate when that will happen. I'll add the RealNVP policies back right when we upgrade our tensorflow version to get the keras compatibility with the RealNVP bijectors.

> I assume that I have to pretrain a lower level policy first. How do I freeze the low level weights in the next step and how can I add a high level policy on top?

You're correct, in the paper, we first train the low-level policies, and then freeze the weights and train the higher-level policy. In practice, the weights are frozen just by embedding the low-level policy in the new high-level environment in between the high-level policy actions and the environment step, and removing those weights from the training graph (or more accurately never adding them to the training graph). We used the [HierarchyProxyEnv](https://github.com/haarnoja/sac/blob/master/sac/envs/hierarchy_proxy_env.py) to do the embedding.

> In the paper you mentioned a set of K reward functions. Where can I define the reward set ?

Right now the rewards have to be manually incorporated in the environments. Here's an example of a low-level policy: https://github.com/haarnoja/sac/blob/master/sac/envs/multi_direction_env.py#L110. And here's an example of a high-level policy: https://github.com/haarnoja/sac/blob/master/sac/envs/cross_maze_ant_env.py#L15

Hope that helps. I'll close this for now since technically it's not an issue in this repo. Feel free to open a new one in the sac repo if you need help.
